drivers/soundwire/stream.c

```diff
@@ -1456,15 +1464,15 @@ static int _sdw_prepare_stream(struct sdw_stream_runtime *stream,
 		if (ret < 0) {
 			dev_err(bus->dev, "Prepare port(s) failed ret = %d\n",
 				ret);
+			stream->state = state;
 			return ret;
```

Comment:
I wonder why this is not jumping to `restore_params`?

Reply:
We didn't change bus->params at this point yet.